releng/release-prep.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+set -e
+
+press_enter() {
+    read -s -p "Press enter to continue"
+    echo
+}
+
+RELEASE_PROMPT="
+Before making the release, make sure that -
+- [ ] the new version number depicts the release accurately
+- [ ] if new features have been added, the MINOR version is bumped
+- [ ] if backwards-incompatible changes have been made, the MINOR version is bumped
+- [ ] if only backwards-compatible bug fixes have been made, the PATCH version is bumped
+- [ ] the changes and the version number has been vetted by project maintainers and stakeholders
+"
+
+echo "$RELEASE_PROMPT"
+press_enter
+
+echo "Fetching latest release tag from GitHub"
+CURRENT_VERSION=$(curl --silent "https://api.github.com/repos/datawire/ambassador/releases/latest" | jq -r '.tag_name')
+
+echo
+echo "Current version: ${CURRENT_VERSION}"
+echo
+git log --pretty=oneline --abbrev-commit ${CURRENT_VERSION}^..
+echo
+echo "^ these changes have been made since the last release, pick the right version number accordingly"
+read -p "Enter new version: " DESIRED_VERSION
+echo
+
+# Update version number in docs/index.html
+echo "Found the following occurrences of $CURRENT_VERSION in ./docs/index.html"
+echo
+grep -C 2 ${CURRENT_VERSION} docs/index.html
+echo
+
+echo "Replacing $CURRENT_VERSION with $DESIRED_VERSION in ./docs/index.html"
+sed -i -e "s/$CURRENT_VERSION/$DESIRED_VERSION/g" docs/index.html
+echo
+git diff docs/index.html
+echo
+echo "^ this is the 'git diff' for docs/index.html"
+press_enter
+
+# Update release notes and changelog now
+RELEASE_NOTES_TEMPLATE="Delete everything from this template that you do not want to be in the template, including this line.
+
+### Major changes:
+- Feature: <insert feature description here>
+- Bugfix: <insert bugfix description here>
+
+### Minor changes:
+- Feature: <insert feature description here>
+- Bugfix: <insert bugfix description here>"
+
+RELEASE_NOTES=$(echo "${RELEASE_NOTES_TEMPLATE}" | vipe)
```

Comment:
So. Editors. Editors suck, man. 😄 Here's the way that seems to be the best chance of avoiding screaming users filing bugs every day because they use Emacs instead of vi, or vi instead of Emacs, or back-in-my-day-we-used-ed-and-it's-good-enough-for-me! :wink:

```
echo "${RELEASE_NOTES_TEMPLATE}" > tmpfile

if ! ${EDITOR:-vi} tmpfile; then
    exit
fi

RELEASE_NOTES=$(cat tmpfile)
rm -f tmpfile
```

except the `tmpfile` needs to actually be a proper temporary file and you should probably use `trap 0` to clean it up.

Also I wouldn't make the user delete that template line. Use `sed` to strip it out maybe?

Reply:
(Admittedly the idea of `vipe` is really really cool. But it's not installed on my system, so this whole thing will fail...)

Reply:
Oh @containscafeine -- please update `RELEASING.md` as part of landing this, too!

Reply:
Totally makes sense, fixed.

Reply:
Updated RELEASING.md